librz/core/cbin.c

```diff
@@ -653,33 +653,29 @@ static bool io_create_mem_map(RzIO *io, RzBinSection *sec, ut64 at) {
 	bool reused = false;
 	ut64 gap = sec->vsize - sec->size;
 	char *uri = rz_str_newf("null://%" PFMT64u, gap);
+	RzIOMap *iomap = NULL;
 	RzIODesc *desc = findReusableFile(io, uri, sec->perm);
 	if (desc) {
-		RzIOMap *map = rz_io_map_get(io, at);
-		if (!map && gap) {
-			rz_io_map_add_batch(io, desc->fd, desc->perm, 0LL, at, gap);
-		}
+		iomap = rz_io_map_add_batch(io, desc->fd, desc->perm, 0LL, at, gap);
```

Comment:
Minor difference here, new maps are always created. It's the responsibility of the bin plugin to handle possible overlaps. Overlaps can also be desired and other io maps from completely different files can be here as well so we should not try to be smarter than we are.

Reply:
This is going to affect only the "bss" sections (or anyway, any sections where vsize > size), right? Do you have any real example where the old behaviour would fail? Just curious for future testing as well, because in theory I can see what you mean and why it could be a problem.

Reply:
@ret2libc see these files as an example. This is a common situation in raw object files: https://github.com/rizinorg/rizin/issues/815

Reply:
Yes, to be exact, it affects sections where `vsize > psize` in which case you get two mappings, one `fmap.<name>` (or none if `psize == 0`) and `mmap.<name>` with 0s.
The check here is for when an `mmap.` one is about to be created. Before, it was skipped if there was another map with a **lower priority** that contained the vaddr (only the **start address**) of the new mmap.

That means:
* The previous behavior would still overlap the .text section with 0s in this example, iff .bss has higher prio:
    ```
    [.bss                         ]
                     [.text                           ]
    ```

* The current behavior's overlaps are more or less unnoticeable if .text has higher prio than .bss, which is the case for example in `eofarch.c-gcc-arm-O3.o` in the linked issue. But I don't think that is something we can rely on since object files are a special case anyway where there are no segments and we fall back to mapping sections, which are actually not meant to be mapped, so priority probably doesn't have much meaning there. (priority is taken from the order they appear in the binary, earlier is higher prio). So #815 is valid and should be fixed insided the elf plugin imo.